include/boost/geometry/extensions/random/uniform_point_distribution.hpp

```diff
@@ -0,0 +1,155 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Tinko Bartels, Berlin, Germany.
+
+// Contributed and/or modified by Tinko Bartels,
+//   as part of Google Summer of Code 2019 program.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_RANDOM_UNIFORM_POINT_DISTRIBUTION_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_RANDOM_UNIFORM_POINT_DISTRIBUTION_HPP
+
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_envelope_rejection.hpp>
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_linear.hpp>
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_point_distribution_discrete.hpp>
+#include <boost/geometry/extensions/random/strategies/cartesian/uniform_point_distribution_box.hpp>
+#include <boost/geometry/extensions/random/strategies/spherical/uniform_inverse_transform_sampling.hpp>
+
+namespace boost { namespace geometry { namespace random
+{
+
+template
+<
+    typename DomainGeometry,
+    typename Point = typename geometry::point_type<DomainGeometry>::type,
+    typename Strategy =
+        typename strategy::uniform_point_distribution::services::default_strategy
+            <
+                Point,
+                DomainGeometry
+            >
```

Comment:
This is not how `default_strategy` would typically be used. It should define `type` member type and be used like this:

    typename xxx::services::default_strategy<X, Y, Z>::type